app/controllers/admin/grove_monitor_controller.rb

```diff
@@ -15,8 +15,23 @@ def show
     authorize(:grove_monitor, :show?)
   end
 
+  def update
+    student = Student.find(params[:id])
+    if student.at_school
+      student.update(student_params)
+    else
+      student.update(student_params)
+    end
+    redirect_to admin_grove_monitor_all_path
+    authorize(:grove_monitor, :index?)
```

Comment:
@brianrip I'm taking care of this, but authorizations should ALWAYS come BEFORE you modify a resource. otherwise, there is no point. the point it to prevent illicit modification of data from those who do not own it.